packages/worker/src/worker.ts

```diff
@@ -1648,6 +1643,38 @@ export class Worker {
   }
 }
 
+export interface WorkflowBundleWithSourceMap {
+  code: string;
+  sourceMap: RawSourceMap;
+  filename: string;
+}
+
+export function parseWorkflowCode(code: string): WorkflowBundleWithSourceMap {
+  const sourceMappingUrlDataRegex = /\s*\n[/][/][#]\s+sourceMappingURL=data:(?:[^,]*;)base64,([0-9A-Za-z+/=]+)\s*$/;
+  const sourceMapMatcher = code.match(sourceMappingUrlDataRegex);
+  if (!sourceMapMatcher) throw new Error("Can't extract inlined source map from the provided Workflow Bundle");
+
+  const sourceMapJson = Buffer.from(sourceMapMatcher[1], 'base64').toString();
+  const sourceMap: RawSourceMap = JSON.parse(sourceMapJson);
+  const filename = path.join(process.cwd(), sourceMap.file);
+
+  // Preloading the script makes breakpoints significantly more reliable and more responsive
+  const script = new vm.Script(code, { filename });
+  const context = vm.createContext({});
+  try {
+    script.runInContext(context);
+  } catch (e) {
+    // Context has not been properly configured, so eventual errors are possible. Just ignore at this point
+  }
+  preloadedScriptAndContext = [script, context];
+
+  return { code, sourceMap, filename };
+}
+
+// Keep these objects from GC. Otherwise the debugger risks source mapping resolution errors
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+let preloadedScriptAndContext: [vm.Script, unknown];
```

Comment:
What happens if multiple bundles are loaded?
Can we store this in a worker attribute?

Reply:
Moved this to a 10 seconds timer. That's well enough for the debugger to parse the source and fully attach to the node process.